### PR TITLE
[Tachyon-1267] Javadoc thrift definitions

### PR DESCRIPTION
**Pull request by ShaoshanLiu**

### CODE REVIEW

**Comment by jsimsa** [comment]
@ShaoshanLiu  thank you for the PR, could you also please regenerate the Thrift Java files, thanks

File: common/src/thrift/block_master.thrift [hunk 1]
```diff
@@ -15,49 +15,69 @@ struct WorkerInfo {
 
 service BlockMasterService extends common.TachyonService {
 
-  // Tachyon Client API
-
-  /*
+  /**
    * Returns the block information for the given block id.
+   * @param blockId
```
Comment (calvinjia) [inline common/src/thrift/block_master.thrift]: Could you complete the javadocs?

Comment (ShaoshanLiu) [inline common/src/thrift/block_master.thrift]: you mean a detailed description of what the param is and what it throws?

Comment (jsimsa) [inline common/src/thrift/block_master.thrift]: I would suggest to remove the `@param`, `@throws` and `@return` annotations altogether ... the reason for that is that Thrift is not smart enough to respect these annotations and it ends up generating its own annotations ... the generated Java sources then end up having duplicate annotations ... for instance:

```
+    /**
+     * Returns the block information for the given block id.
+     * @param blockId
+     * @return the block information for the given block id
+     * @throws TachyonTException
+     * 
+     * @param blockId
+     */
```

which is not great

Comment (ShaoshanLiu) [inline common/src/thrift/block_master.thrift]: so do you just want to leave the description itself, removing all return, param, and throws? Let me know. @jsimsa @calvinjia

Comment (jsimsa) [inline common/src/thrift/block_master.thrift]: that's what I am thinking ... @calvinjia what do you think?

Comment (ShaoshanLiu) [inline common/src/thrift/block_master.thrift]: @calvinjia, any comments?

**Comment by ShaoshanLiu** [comment]
@calvinjia @jsimsa guys, this has been sitting here for a few days, do we have a decision of what to do next? 

thanks!

Shaoshan

File: common/src/thrift/block_master.thrift (hunk 1 above)
Comment (calvinjia) [inline common/src/thrift/block_master.thrift]: @ShaoshanLiu @jsimsa 
I think we can do something like

```
/**
   * Marks a file as completed.
   */
  void completeFile( /** the id of the file */ 1: i64 fileId, /** the options for this operation */ 2: CompleteFileTOptions options)
    throws (1: exception.TachyonTException e)
```

**Comment by calvinjia** [comment]
@ShaoshanLiu I think there's a way to leverage thrift's auto generated javadocs, the formatting is a bit weird though.

File: common/src/thrift/block_master.thrift (hunk 1 above)
Comment (jsimsa) [inline common/src/thrift/block_master.thrift]: nice, if we do that, let's have one argument per line

Comment (ShaoshanLiu) [inline common/src/thrift/block_master.thrift]: cool, I will have this fixed.

Shaoshan

**Comment by ShaoshanLiu** [comment]
never mind, seems that I got some other commits during merge, let me clean up and reopen this.

**Pull request closed without merging**